src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,650 @@
+
+We see the differences in ``union``::
+
+    sage: H.<x,y> = HyperplaneArrangements(QQ)
+    sage: H1.<x,y> = OrderedHyperplaneArrangements(QQ)
+    sage: A = H([1,2,3], [0,1,1], [0,1,-1], [1,-1,0], [1,1,0])
+    sage: B = H([1,1,1], [1,-1,1], [1,0,-1])
+    sage: C = A.union(B)
+    sage: A1 = H1(A); B1 = H1(B); C1 = A1.union(B1)
+    sage: [C1.hyperplanes().index(h) for h in C.hyperplanes()]
+    [0, 5, 6, 1, 2, 3, 7, 4]
+
+Also in ``cone``::
+
+    sage: # needs sage.combinat
+    sage: a.<x,y,z> = hyperplane_arrangements.semiorder(3)
+    sage: H.<x,y,z> = OrderedHyperplaneArrangements(QQ)
+    sage: a1 = H(a)
+    sage: b = a.cone(); b1 = a1.cone()
+    sage: [b1.hyperplanes().index(h) for h in b.hyperplanes()]
+    [0, 2, 4, 6, 1, 3, 5]
+
+And in ``restriction``::
+
+    sage: # needs sage.graphs
+    sage: A.<u, x, y, z> = hyperplane_arrangements.braid(4)
+    sage: L.<u, x, y, z> = OrderedHyperplaneArrangements(QQ)
+    sage: A1 = L(A)
+    sage: H = A[0]; H
+    Hyperplane 0*u + 0*x + y - z + 0
+    sage: A.restriction(H)
+    Arrangement <x - z | u - x | u - z>
+    sage: A1.restriction(H)
+    Arrangement <x - z | u - x | u - z>
+    sage: A1.restriction(H, repetitions=True)
+    Arrangement of 5 hyperplanes of dimension 3 and rank 2
+
+AUTHORS:
+
+- Enrique Artal (2023-12): initial version
+
+This module add some features to the *unordered* one for some
+properties which depend on the order.
+"""
+
+# *****************************************************************************
+#       Copyright (C) 2023 Enrique Artal <artal@unizar.es>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  http://www.gnu.org/licenses/
+# *****************************************************************************
+
+from sage.geometry.hyperplane_arrangement.arrangement import HyperplaneArrangementElement
+from sage.geometry.hyperplane_arrangement.arrangement import HyperplaneArrangements
+from sage.geometry.hyperplane_arrangement.hyperplane import Hyperplane
+from sage.matrix.constructor import matrix, vector
+from sage.misc.misc_c import prod
+from sage.groups.free_group import FreeGroup
+from sage.rings.integer_ring import ZZ
+from sage.rings.qqbar import QQbar
+from sage.schemes.curves.plane_curve_arrangement import AffinePlaneCurveArrangements
+from sage.schemes.curves.plane_curve_arrangement import ProjectivePlaneCurveArrangements
+
+
+class OrderedHyperplaneArrangementElement(HyperplaneArrangementElement):
+    """
+    An ordered hyperplane arrangement.
+
+    .. WARNING::
+
+        You should never create
+        :class:`OrderedHyperplaneArrangementElement` instances directly,
+        always use the parent.
+    """
+
+    def __init__(self, parent, hyperplanes, check=True, backend=None):
```

Comment:
```suggestion
    """
    def __init__(self, parent, hyperplanes, check=True, backend=None):
```
We generally do not have this blankline.